samtranslator/model/api/api_generator.py

```diff
@@ -515,22 +515,29 @@ def _construct_api_domain(self, rest_api):
         record_set_group = None
         if self.domain.get("Route53") is not None:
             route53 = self.domain.get("Route53")
-            if route53.get("HostedZoneId") is None and route53.get("HostedZoneName") is None:
+            if isinstance(route53, dict):
```

Comment:
nit: we can just add `if not isintance(route53, dict): raise ...` so that we don't need to change any of below?